test/name_bazaar/smart_contract_tests.cljs

```diff
@@ -35,55 +37,467 @@
+                 (testing
+                     "Can place a proper bid"
+                   (is (tx-sent? (<! (auction-offering/bid! ss
+                                                            {:offering/address offering}
+                                                            {:value (web3/to-wei 0.1 :ether)
+                                                             :from (state/my-address 1)})))))
+                 (testing
+                     "Next bid should respect min-bid-increase"
+                   (is (tx-failed? (<! (auction-offering/bid! ss
+                                                              {:offering/address offering}
+                                                              {:value (web3/to-wei 0.11 :ether)
+                                                               :from (state/my-address 2)})))))
+                 (testing
+                     "Correct increase of the bid is accepted"
+                   (is (tx-sent? (<! (auction-offering/bid! ss
+                                                            {:offering/address offering}
+                                                            {:value (web3/to-wei 0.2 :ether)
+                                                             :from (state/my-address 2)})))))
+                 (testing
+                     "Arbitrary increase of the bid is ok"
+                   (is (tx-sent? (<! (auction-offering/bid! ss
+                                                            {:offering/address offering}
+                                                            {:value (web3/to-wei 0.33 :ether)
+                                                             :from (state/my-address 3)})))))
+                 (testing
+                     "State of the auction offering is correct"
+                   (is (= {:auction-offering/min-bid-increase 100000000000000000
+                           :auction-offering/winning-bidder (state/my-address 3)
+                           :auction-offering/bid-count 3}
+                          (select-keys (last (<! (auction-offering/get-auction-offering ss
+                                                                                        offering)))
+
+                                       [:auction-offering/min-bid-increase
+                                        :auction-offering/winning-bidder
+                                        :auction-offering/bid-count]))))
+
+                 (testing
+                     "Can't finalize the auction prematurely"
+                   (is (tx-failed? (<! (auction-offering/finalize! ss
+                                                                   {:offering/address offering
+                                                                    :offering/transferPrice true}
+                                                                   {:from (state/my-address 0)})))))
+                 (web3-evm/increase-time!
```

Comment:
Why not to use async version? `cljs-web3.async.evm/increase-time!`

Reply:
Async version also works, but this also triggers this nasty bug with too many assertions in one go block. Callback version is ugly, but I would prefer not to break this into several scenarios.

Reply:
I see, okay cool